services-geojson/src/commonMain/kotlin/org/maplibre/geojson/model/Polygon.kt

```diff
@@ -0,0 +1,104 @@
+import org.maplibre.geojson.utils.json
+import kotlin.jvm.JvmStatic
+
+/**
+ * Model representing a GeoJSON Polygon. Polygons are build by an outer [LineString] ring and
+ * zero or more inner [LineString] rings.
+ *
+ * Every ring (inner, and outer) are a closed linear line rings. This means, that minimum four points are required,
+ * and the first and last points must be identical.
+ *
+ * See [Polygon specification](https://datatracker.ietf.org/doc/html/rfc7946#section-3.1.6) for more details.
+ *
+ * @param outerLineStringRing the outer [LineString] ring of this Polygon
+ * @param holeLineStringRings the inner [LineString] rings of this Polygon. Or empty list if no inner holes exists
+ * @param boundingBox the BoundingBox of this Polygon
+ * @throws IllegalArgumentException if one of the provided [LineString] is not valid linear rings
+ */
+@Serializable(with = PolygonSerializer::class)
+@SerialName("Polygon")
+data class Polygon(
+    /**
+     * The outer [LineString] ring of this Polygon.
+     */
+    val outerLineStringRing: LineString,
+
+    /**
+     * The inner [LineString] rings of this Polygon. Or empty list if no inner holes exists.
+     */
+    val holeLineStringRings: List<LineString> = emptyList(),
+
+    /**
+     * The [BoundingBox] of this Polygon.
+     */
+    override val boundingBox: BoundingBox? = null,
+) : Geometry {
+
+    /**
+     * Constructor to create a Polygon with outer ring.
+     *
+     * @param outer the outer LineString ring
+     */
+    constructor(outer: LineString) : this(outer, emptyList(), null)
+
+    /**
+     * Constructor to create a Polygon with outer and inner rings.
+     *
+     * @param outer the outer LineString ring
+     * @param inner the inner LineString rings
+     */
+    constructor(outer: LineString, inner: List<LineString>) : this(outer, inner, null)
+
+    /**
+     * Check on initializing, if this Polygon is valid.
+     *
+     * @throws IllegalArgumentException if the [Polygon] is invalid
+     */
+    init {
+        outerLineStringRing.requireLinearRing()
+        holeLineStringRings.forEach { line -> line.requireLinearRing() }
+    }
+
+    /**
+     * Converts this [Polygon] to its GeoJSON representation, as String.
+     *
+     * @return a String that contains JSON
+     */
+    override fun toJson(): String = json.encodeToString(this)
+
+    /**
+     * Check if given [LineString] is a linear ring.
+     *
+     * That means:
+     * 1. Minimum four points are contained.
+     * 2. First and last points are identical.
+     *
+     * @throws IllegalArgumentException if the [LineString] is not a linear ring
+     */
+    private fun LineString.requireLinearRing() {
+        require(points.size >= 4) {
+            "LinearString for Polygon rings need to be made up of 4 or more Points."
+        }
+
+        require(points.first() == points.last()) {
+            "LinearString for Polygon rings require first and last Point to be identical."
+        }
```

Comment:
For more informative error message (i.e. which ring exactly is wrong):

```suggestion
    private fun LineString.requireLinearRing(name: String, index: Int?) {
        require(points.size >= 4) {
            "${getRingName}: LineString for Polygon rings need to be made up of 4 or more Points."
        }

        require(points.first() == points.last()) {
            "${getRingName}: LineString for Polygon rings require first and last Point to be identical."
        }
    
    private fun getRingName(name: String, index: Int?): String {
        val indexString = index?.let { " at index $it" } ?: ""
        return "$name$indexString"
    }
```

(also, typo: "LinearString" -> "LineString")